File: app/src/main/java/info/firozansari/espressoexamples/adapters/PagerAdapter.java
package info.firozansari.espressoexamples.adapters;
import android.app.Fragment;
import android.app.FragmentManager;
import android.support.v13.app.FragmentStatePagerAdapter;

import info.firozansari.espressoexamples.fragments.PagerFragment;


public class PagerAdapter extends FragmentStatePagerAdapter {
    public String[] mStrings;


    public PagerAdapter(FragmentManager fm) {
        super(fm);
    }

    public PagerAdapter(FragmentManager fm, String[] strings) {
        super(fm);
        mStrings = strings;
    }

    @Override
    public Fragment getItem(int position) {
        return new PagerFragment().newInstance(mStrings[position]);
    }

    @Override
    public int getCount() {
        return mStrings.length;
    }
}

File: app/src/main/java/info/firozansari/espressoexamples/adapters/PagerAdapter.kt
package info.firozansari.espressoexamples.adapters

import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager
import androidx.fragment.app.FragmentStatePagerAdapter
import info.firozansari.espressoexamples.fragments.PagerFragment

class PagerAdapter : FragmentStatePagerAdapter {
    lateinit var mStrings: Array<String>

    constructor(fm: FragmentManager?) : super(fm!!) {}
    constructor(fm: FragmentManager?, strings: Array<String>) : super(fm!!) {
        mStrings = strings
    }

    override fun getItem(position: Int): Fragment {
        return PagerFragment.newInstance(mStrings[position])
    }

    override fun getCount(): Int {
        return mStrings.size
    }
}